### DISCUSSION TITLE
Unable to start commissioner on border router with cli

### DISCUSSION THREAD

**Discussion by michaelAeOST** [openthread/openthread#8504] [Q&A] [answered]
I set up a border router with a Raspberry Pi 4 and a SLWSTK6000B from Silicon Labs according to the description. It works fine so far. Then I tried to start the commissioner using the commissioner-cli.

Unfortunately I was not able to start the commissioner. There is following error.
**IO_ERROR: NET - Reading information from the socket failed**

According to the description I may chosen the wrong IP address or port. But which one is the correct IP address and how can I find out the port? I selected the port according to the Service Browser app on my smart phone where I can find the _meshcop._udp service of my border router.

Below are the commands I used to start commissioner.

Thank you for the support.

```
otbr@raspberrypi:~ $ dns-sd -G v4/v6 raspberrypi.local
DATE: ---Thu 08 Dec 2022---
 7:41:42.607  ...STARTING...
Timestamp     A/R    Flags if Hostname                               Address                                      TTL
 7:41:42.607  Add 40000003  2 raspberrypi.local.                     FD11:1111:1122:2222:E65F:01FF:FEBB:244D%<0>  120
 7:41:42.608  Add 40000003  2 raspberrypi.local.                     2001:0620:01C0:1016:E65F:01FF:FEBB:244D%<0>  120
 7:41:42.608  Add 40000003  2 raspberrypi.local.                     FD11:1111:1122:2222:CED3:CEE8:03DD:CC9B%<0>  120
 7:41:42.608  Add 40000003  2 raspberrypi.local.                     2001:0620:01C0:1016:D013:CD2A:31AE:2F78%<0>  120
 7:41:42.608  Add 40000003  2 raspberrypi.local.                     FE80:0000:0000:0000:E65F:01FF:FEBB:244D%eth0 120
 7:41:42.608  Add 40000002  2 raspberrypi.local.                     146.136.23.190                               120
otbr@raspberrypi:~ $ sudo ot-ctl dataset
Active Timestamp: 1
Channel: 16
Channel Mask: 0x07fff800
Ext PAN ID: 1111111122222222
Mesh Local Prefix: fdbf:9f21:b944:e3b9::/64
Network Key: 00112233445566778899aabbccddeeff
Network Name: EsaThread
PAN ID: 0x1234
PSKc: 32637ae3bcb8338b2fce09c0daad436b
Security Policy: 672 onrc
Done
otbr@raspberrypi:~ $ commissioner-cli
   ____  ______                                   _           _                          ________    ____
  / __ \/_  __/   _________  ____ ___  ____ ___  (_)_________(_)___  ____  ___  _____   / ____/ /   /  _/
 / / / / / /_____/ ___/ __ \/ __ `__ \/ __ `__ \/ / ___/ ___/ / __ \/ __ \/ _ \/ ___/  / /   / /    / /
/ /_/ / / /_____/ /__/ /_/ / / / / / / / / / / / (__  |__  ) / /_/ / / / /  __/ /     / /___/ /____/ /
\____/ /_/      \___/\____/_/ /_/ /_/_/ /_/ /_/_/____/____/_/\____/_/ /_/\___/_/      \____/_____/___/


> config set pskc 32637ae3bcb8338b2fce09c0daad436b
[done]
> start FD11:1111:1122:2222:E65F:01FF:FEBB:244D 49153
IO_ERROR: NET - Reading information from the socket failed
[failed]
>
```

**Comment by jwhui** [comment, marked as answer]
The IPv6 address and port should match what OTBR is advertising as the `_meshcop._udp` service via mDNS. See the [Discover your network](https://openthread.io/guides/border-router/external-commissioning/cli#discover_your_network) section of the [OT Commissioner guide](https://openthread.io/guides/border-router/external-commissioning/cli).

**Reply by michaelAeOST** [reply]
> The IPv6 address and port should match what OTBR is advertising as the `_meshcop._udp` service via mDNS. See the [Discover your network](https://openthread.io/guides/border-router/external-commissioning/cli#discover_your_network) section of the [OT Commissioner guide](https://openthread.io/guides/border-router/external-commissioning/cli).

Thank you for the fast response. It works if I use the IP and port reported for _meshcop._udp on my smartphone. Now I tried another device (SLWSTK6000B from Silicon Labs using the ot-cli-mtd demo) to join the network according the [Commission the joiner](https://openthread.io/guides/border-router/external-commissioning/cli#commission_the_joiner). But this action fails. What is wrong? See my setup of the commissioner and the joiner below.

Commissioner:
```
> config set pskc 630d21e58fc56f1f9a172b559445b182
[done]
> start fdee:5dad:cf44:b111:e65f:1ff:febb:244d 49153
[done]
> active
true
[done]
> joiner enableall meshcop J01NU5
[done]
```

Joiner:
```
> ifconfig up
Done
> joiner start J01NU5
Done
> [N] MeshForwarder-: Failed to send IPv6 UDP msg, len:56, chksum:b593, ecn:no, to:None, sec:no, error:Abort, prio:net
[N] MeshForwarder-:     src:[fe80:0:0:0:4c5d:9501:74a0:7b05]:19788
[N] MeshForwarder-:     dst:[ff02:0:0:0:0:0:0:2]:19788
Join failed [NotFound]
```

**Reply by jwhui** [reply]
It appears that your joiner is not able to transmit messages, but it's not clear to me why. You can try using the `factoryreset` command before the `ifconfig up` call. If that does not help, then I would reach out to Silicon Labs for support.

**Reply by michaelAeOST** [reply]
Thank you. Now it works.
